src/TrajectoryFollower.cpp

```diff
@@ -165,25 +165,26 @@ FollowerStatus TrajectoryFollower::traverseTrajectory(Motion2D &motionCmd, const
     }
 
     /*
-        Here we need to differentiate whether the DriveMode::ModeTurnOnTheSpot is set by the 
+        Here we need to differentiate whether the DriveMode::ModeTurnOnTheSpot is set by the
         automatic point turn feature of trajectory follower or the DriveMode::ModeTurnOnTheSpot
-        is actually required by the planner as part of the planned trajectory 
-    */  
+        is actually required by the planner as part of the planned trajectory
+    */
 
-    if (trajectory.driveMode == DriveMode::ModeTurnOnTheSpot && automaticPointTurn == false) 
+    if (trajectory.driveMode == DriveMode::ModeTurnOnTheSpot && automaticPointTurn == false)
     {
         double actualHeading = robotPose.getYaw();
         double targetHeading = trajectory.goalPose.orientation;
 
         if (actualHeading < 0)
-            actualHeading = 2*M_PI + actualHeading; 
-
+            actualHeading = 2*M_PI + actualHeading;
+        if (targetHeading < 0)
+            targetHeading = 2*M_PI + targetHeading;
         double error       = actualHeading - targetHeading;
```

Comment:
If your problem is the classical "oscillating around 2*PI", the better fix is to normalize `actualHeading  - targetHeading` within [-PI, PI]